core/src/main/scala/org/genivi/sota/core/user_profile/UserProfileClient.scala

```diff
@@ -63,4 +71,33 @@ class UserProfileClient
     }
   }
 
+  override def getRepoId
+    (ns: Namespace)
+    (implicit ec: ExecutionContext): Future[RepoId] = {
+
+    val request = HttpRequest(
+      method = HttpMethods.GET,
+      uri = baseUri.withPath(api.path / "users" / ns.get))
+
+    http.singleRequest(request).flatMap { response =>
+      response.status match {
+        case StatusCodes.NotFound =>
+          log.error(MissingUser.desc)
+          FastFuture.failed(MissingUser)
+        case x if x.isSuccess() =>
+          Unmarshal(response.entity).to[RepoIdWrapper].flatMap { _.repoId match {
+            case Some(repoId) =>
+              FastFuture.successful(repoId)
+            case None =>
+              val err = "user does not have a repo yet"
+              log.error(err)
+              FastFuture.failed(new Exception(err))
```

Comment:
It would be good if this was also a proper error, `MissingRepo` instead of a 500.